src/realm/table.cpp

```diff
@@ -291,6 +291,36 @@ void Table::insert_column_link(size_t col_ndx, DataType type, StringData name, T
     get_descriptor()->insert_column_link(col_ndx, type, name, target, link_type); // Throws
 }
 
+size_t Table::add_column_list(DataType type, StringData name)
+{
+    size_t ndx = get_column_count();
+    insert_column_list(ndx, type, name);
+    return ndx;
+}
+
+void Table::insert_column_list(size_t column_ndx, DataType type, StringData name)
+{
+    DescriptorRef subdesc;
+    insert_column(column_ndx, type_Table, name, &subdesc);
+    switch (type) {
+        case type_Int:
+        case type_Bool:
+        case type_Float:
+        case type_Double:
+        case type_String:
+        case type_Binary:
+        case type_Timestamp:
+            subdesc->add_column(type, "list");
+            break;
+        case type_OldDateTime:
+        case type_Table:
+        case type_Mixed:
+        case type_Link:
+        case type_LinkList:
+            throw LogicError(LogicError::type_not_supported);
+            break;
+    }
```

Comment:
Will the compiler warn when we add a new type and forget to add it to this list? I don't remember the status of that wish